Add indent and ftplugin files

adriaanzon opened the pull request:
I added indentation for css and javascript manually, instead of relying on Vim's indent/html.vim, because that seemed to work much better.  [othree's html5.vim](https://github.com/othree/html5.vim) also does this, so anthother solution might be to get rid of the css/js indentation rules and recommend the html5.vim plugin for better indentation.

Pug indentation didn't seem to work properly for me, even in standalone files.

I also included a `ftplugin/vue.vim` which sources `ftplugin/html.vim`.  This enables you to use matchit to jump between tag pairs.

posva commented:
Thank you very much for this. I'm currently trying out.
When using html5.vim, do you mean to use the same filetype for vue files?

posva commented:
I tested by setting `ft=html` with html5.vim installed and it works pretty well when you have css, html and javascript (even es6). Your work makes it exactly as with html5.vim when using those languages
BTW You copied my name on those files, you may want to take credit for that before i merge the PR @adriaanzon

adriaanzon commented:
Yes, you would still use the `vue` filetype.  The html5.vim plugin gets used automatically because we're importing it using `runtime`.

posva commented:
Good, to know. I'm happy you could help out 👍 
I wouldn't be able to do it without your help
About the names, do you want to add yours or we merge it this way?

adriaanzon commented:
I will still make some changes, because I just noticed indentation doesn't work for stylus after I tweaked the regex for css.

posva commented:
Ok, didn't have any issue while trying it with or without `{}`

adriaanzon commented:
Did you add `lang="stylus"` to the style tag?

posva commented:
@adriaanzon of course 😄 
Here is the file I used: https://github.com/posva/vue-mdl/blob/develop/test/components/Menu.vue

posva commented:
I found that sometimes it's necessary to manually execute `set ft=vue`

adriaanzon commented:
I changed the order of the `searchpair` statements so stylus indentation. Also added my name at the top :p 

Doesn't vim set the filetype or do you have to execute `set ft=vue` again?

posva commented:
Great!
It should 😕 . I think it happens to me because of vim_restoreview plugin which also restore the ft
